packages/editor/src/components/document-outline/check.js

```diff
@@ -19,7 +19,7 @@ export default function DocumentOutlineCheck( { children } ) {
 		return getGlobalBlockCount( 'core/heading' ) > 0;
 	} );
 
-	if ( hasHeadings ) {
+	if ( ! hasHeadings ) {
```

Comment:
I thought I'd fixed it, but changes probably got lost during the force push.

Props to @t-hamano for pointing it out [initially](https://github.com/WordPress/gutenberg/pull/59209#discussion_r1498497252).